domain/consensus/processes/syncmanager/syncinfo.go

```diff
@@ -54,11 +52,15 @@ func (sm *syncManager) resolveSyncState() (externalapi.SyncState, error) {
 		return externalapi.SyncStateHeadersFirst, nil
 	}
 
-	headerVirtualSelectedParentBlockStatus, err := sm.blockStatusStore.Get(sm.databaseContext, headerVirtualSelectedParentHash)
+	headerTipsPruningPoint, err := sm.consensusStateManager.HeaderTipsPruningPoint()
+	if err != nil {
+		return 0, err
+	}
+	headerTipsPruningPointStatus, err := sm.blockStatusStore.Get(sm.databaseContext, headerTipsPruningPoint)
 	if err != nil {
 		return 0, err
 	}
-	if headerVirtualSelectedParentBlockStatus != externalapi.StatusValid {
+	if headerTipsPruningPointStatus != externalapi.StatusValid {
```

Comment:
Please add comment explaining that this can happen only if we are still in headers first.